pkg/trace/api/dogstatsd.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022-present Datadog, Inc.
+
+package api
+
+import (
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/pkg/trace/log"
+)
+
+// dogstatsdProxyHandler returns a new HTTP handler which will proxy requests to
+// the DogStatsD endpoint in the Core Agent over UDP or UDS (defaulting to UDS
+// if StatsdSocket is set in the *AgentConfig).
+func (r *HTTPReceiver) dogstatsdProxyHandler() http.Handler {
+	if !r.conf.StatsdEnabled {
+		log.Info("DogstatsD disabled in the Agent configuration")
```

Comment:
Is this configuration for `DogstatsD` enabled in general? Or specifically the proxy?

Reply:
I guess both? If `Dogstatsd` isn't enabled in the Agent (i.e. in general), then this endpoint will not be enabled. This isn't a separate configuration just to enable to proxy, if that's what you mean.